### DISCUSSION TITLE
Copilot for Xcode?

### DISCUSSION THREAD

**Discussion by Ben216k** [community/community#6860] [Copilot Conversations] [answered]
Would it be possible to bring GitHub Copilot to Xcode? My concern would be whether or not you can do that with an Xcode extension since Apple is known for limiting this.

**Comment by sancho1952007** [comment]
Currently, Copilot Only Works For VS Code.
But, I'm Sure They Will Add Support For Other Text/Code Editors Also...

**Comment by WestFR** [comment]
Copilot can worked perfectly on Android Studio because it's has a IntelliJ base code source.

And it's already worked on Swift language (on different text / code editors) like this demo : https://twitter.com/_inside/status/1410338216729878530

 XCode is a tricky different, so i think we just wait for this. We don't have access to code source of Github Copilot as well for creating a bridge on it.

**Comment by wangyiwei2015** [comment]
You may try JetBrains AppCode with the Copilot plugin.

**Reply by yonivav** [reply]
did anyone try?

**Reply by mrjavicho** [reply]
I have just tried it out and it's working nicely in AppCode.
I have an on/off relationship with this IDE though, it will be nice to be able to use copilot in XCode.

**Reply by DmytroLisitsyn** [reply]
Works like charm.

**Reply by morphinewan** [reply]
It is sad. With the release of v2022.3, jetbrains are sunsetting the product.  https://blog.jetbrains.com/appcode/2022/12/appcode-2022-3-release-and-end-of-sales-and-support/

**Reply by johnvicinno** [reply]
We ended up making one for our own use. [CoderKit](https://coderkit.ai ) it is free to use and also support [Codeium](https://codeium.com). happy to share with other developers.

**Comment by sardorb3k** [comment]
Copilot works in JetBrains AppCode

**Reply by marciogranzotto** [reply]
AppCode was sunsetted: https://blog.jetbrains.com/appcode/2022/12/appcode-2022-3-release-and-end-of-sales-and-support/

**Reply by Yevhen-Myroshnychenko** [reply]
AppCode is not supported anymore.

**Comment by mickeyl** [comment]
An Xcode Editor Plugin should be workable.

**Comment by alloc33** [comment]
C'mon, XCode doesn't even have properly VIM support... but sometimes it helps, yes - to manage assets 😆... Ok, respect XCode - you started with it...

**Comment by ActuallyHappening** [comment]
https://github.com/intitni/CopilotForXcode - brew package I randomly noticed

**Reply by siginur** [reply]
Somebody tried to use it? How is it?

**Comment by JPwharton** [comment]
I've been using [CodeAI](https://apps.apple.com/us/app/codeai-ai-for-xcode/id6446805484?mt=12) and it works pretty well.  Haven't tested the brew package one yet.

**Reply by johnvicinno** [reply]
We ended up making one for our own use. [CoderKit](https://coderkit.ai/) it is free to use and also support [Codeium](https://codeium.com/). happy to share with other developers

**Comment by Akash1134** [comment, marked as answer]
## 👀  [GitHub Copilot code completion in Xcode is now available in public preview](https://github.com/orgs/community/discussions/141898) 🚀

**Comment by datonefeed** [comment]
It's possible that future updates to Xcode, GitHub Copilot, or third-party tools could bridge this gap, but for now, Xcode doesn't directly support GitHub Copilot.

**Comment by Akash1134** [comment]
We're happy to announce that GitHub Copilot Chat for Xcode is now in Public Preview!

You can now experience AI-powered coding assistance in Xcode! With just a [GitHub account](https://github.com/features/copilot), enable Copilot and get both code completions and in-editor chat to supercharge your workflow.

#### What’s New
✅ Chat view – Ask Copilot for coding help directly within Xcode.
✅ Slash commands – Use shortcuts like /explain for instant insights.
✅ Reference code – Scope chats to specific files for precise assistance.
✅ Multiple conversations – Keep separate threads for different topics.
✅ Chat history management – Revisit past discussions anytime.
✅ Free Tier Access – Get [2,000 completions & 50 chat messages](https://github.com/copilot) per month, just by signing in!

#### 🚀 Get Started

- A [Copilot license](https://github.com/features/copilot) is required.
- Follow the [getting started guide](https://aka.ms/copilotxcode) to set up Copilot in Xcode today!

Try it out and let us know what you think! 🚀
